content/blog/2021/08/2021-08-14-introducing-junit-external-storage.adoc

```diff
@@ -0,0 +1,147 @@
+---
+:layout: post
+:title: "Introducing external storage for JUnit test results"
+:tags:
+- cloud-native
+- pluggable-storage
+- junit
+:author: timja
+:sig: gsoc
+// opengraph: // TODO
+//   image:
+links:
+//   discourse: TODO
+
+---
+
+In common CI/CD use-cases a lot of the space is consumed by test reports.
+This data is stored within `JENKINS_HOME`,
+and the current storage format requires huge overheads when retrieving statistics and, especially, trends.
+In order to display trends, each report has to be loaded and then processed in-memory.
+
+The main purpose of externalising Test Results is to optimize Jenkins performance and storage
+by querying the desired data from external storages.
+
+I'm please to announce that the plugin:junit[JUnit Plugin] external storage is now available for use.
+
+== Getting started
+
+Install your database vendor specific plugin, you can use the Jenkins plugin site to search for it:
+
+https://plugins.jenkins.io/ui/search/?labels=database
+
+e.g. you could install the plugin:database-postgresql[PostgreSQL Database] plugin.
+
+We currently support plugin:database-postgresql[PostgreSQL] or plugin:database-mysql[MySQL], but can support others, just create an issue or send a pull request.
+
+=== From Jenkins UI
+
+Navigate to: Manage Jenkins → Configure System → Junit
+
+In the dropdown select 'SQL Database'
+
+image:/images/post-images/2021/junit-external-storage/junit-sql-config-screen.png[JUnit SQL plugin configuration]
+
+Now configure your Database connection details.
+
+Search for 'Global Database' on the same 'Configure System' page.
+
+Select the database implementation you want to use and click 'Test Connection' to verify Jenkins can connect
+
+image:/images/post-images/2021/junit-external-storage/junit-sql-config-screen.png[JUnit SQL plugin database configuration]
+
+Click 'Save'
+
+=== Configuration as code
+
+If you want to configure the plugin via plugin:configuration-as-code[Configuration as Code] then see the below sample:
+
+[source,yaml]
+----
+unclassified:
+  globalDatabaseConfiguration:
+    database:
+      postgreSQL:
+        database: "jenkins"
+        hostname: "${DB_HOST_NAME}"
+        password: "${DB_PASSWORD}"
+        username: "${DB_USERNAME}"
+        validationQuery: "SELECT 1"
+  junitTestResultStorage:
+    storage: "database"
+----
+
+=== Using the plugin
+
+Now run some builds, here's an example pipeline configuration to get you started if you're just trying out the plugin:
```

Comment:
https://github.com/jglick/simple-maven-project-with-tests might be more helpful?